plugins/json/src/main/resources/struts-plugin.xml

```diff
@@ -5,8 +5,11 @@
 	"http://struts.apache.org/dtds/struts-2.5.dtd">
 
 <struts>
-    <bean class="org.apache.struts2.json.JSONWriter"/>
-    <bean class="org.apache.struts2.json.JSONUtil" static="true"/>
+    <bean type="org.apache.struts2.json.JSONWriter" name="json" class="org.apache.struts2.json.DefaultJSONWriter"
+          scope="prototype"/>
+    <constant name="struts.json.writer" value="json"/>
+
+    <bean class="org.apache.struts2.json.JSONUtil" scope="prototype"/>
```

Comment:
Hm... but the class is using static cache maps which means it can a `singleton`, right?

Reply:
You are right. Actually I had to make it prototype because `DefaultJSONWriter` is prototype. `JSONInterceptor` and `JSONResult` need `JSONUtil` to being injected. If I make it singleton, then it fetches only one instance of `DefaultJSONWriter` and unfortunately, currently, `DefaultJSONWriter` cannot work correctly shared between concurrent threads :(

Reply:
💯

Reply:
I added following to `plugin.xml` keeping history ;)
```xml
<!-- TODO: Make DefaultJSONWriter thread-safe to remove "prototype"s -->
```